packages/gatsby/cache-dir/loader.js

```diff
@@ -130,82 +115,60 @@ const queue = {
 
     const mountOrderBoost = 1 / mountOrder
     mountOrder += 1
+
+    function enqueueResource(resourceName) {
+      if (!resourceName) return
+      if (!resourcesCount[resourceName]) {
+        resourcesCount[resourceName] = 1 + mountOrderBoost
+      } else {
+        resourcesCount[resourceName] += 1 + mountOrderBoost
+      }
+
+      // Before adding, checking that the resource isn't either
+      // already queued or been downloading.
+      if (hasFetched[resourceName] || resourcesArray.includes(resourceName))
+        return
+
+      resourcesArray.unshift(resourceName)
+    }
+
     // console.log(
-    // `enqueue "${path}", mountOrder: "${mountOrder}, mountOrderBoost: ${mountOrderBoost}`
+    // `enqueue "${path}", mountOrder: ${mountOrder}, mountOrderBoost: ${mountOrderBoost}`
     // )
 
     // Add to path counts.
-    if (!pathCount[path]) {
-      pathCount[path] = 1
-    } else {
-      pathCount[path] += 1
-    }
+    // if (!pathCount[path]) {
+    //   pathCount[path] = 1
+    // } else {
+    //   pathCount[path] += 1
+    // }
 
-    // Add path to queue.
-    if (!queue.has(path)) {
-      pathArray.unshift(path)
-    }
+    // // Add path to queue.
+    // if (!queue.has(path)) {
+    //   pathArray.unshift(path)
+    // }
 
-    // Sort pages by pathCount
-    pathArray.sort(sortPagesByCount)
+    // // Sort pages by pathCount
+    // pathArray.sort(sortPagesByCount)
 
     // Add resources to queue.
     const page = findPage(path)
-    if (page.jsonName) {
-      if (!resourcesCount[page.jsonName]) {
-        resourcesCount[page.jsonName] = 1 + mountOrderBoost
-      } else {
-        resourcesCount[page.jsonName] += 1 + mountOrderBoost
-      }
-
-      // Before adding, checking that the JSON resource isn't either
-      // already queued or been downloading.
-      if (
-        resourcesArray.indexOf(page.jsonName) === -1 &&
-        !resourceStrCache[page.jsonName]
-      ) {
-        resourcesArray.unshift(page.jsonName)
-      }
-    }
-    if (page.componentChunkName) {
-      if (!resourcesCount[page.componentChunkName]) {
-        resourcesCount[page.componentChunkName] = 1 + mountOrderBoost
-      } else {
-        resourcesCount[page.componentChunkName] += 1 + mountOrderBoost
-      }
-
-      // Before adding, checking that the component resource isn't either
-      // already queued or been downloading.
-      if (
-        resourcesArray.indexOf(page.componentChunkName) === -1 &&
-        !resourceStrCache[page.jsonName]
-      ) {
-        resourcesArray.unshift(page.componentChunkName)
-      }
-    }
 
+    enqueueResource(page.jsonName)
+    enqueueResource(page.componentChunkName)
```

Comment:
do we need to do the layout here as well? wondering if that was missed in the original PR?